articles/how-and-when-to-apply-session-storage-with-JavaScript/index.md

```diff
@@ -0,0 +1,224 @@
+
+    const car = {
+        brand: brand,
+        price: price,
+    }
+
+    window.sessionStorage.setItem('car',JSON.stringify(car));  
+    //converting object to string
+}
+```
+
+The `main.js` also has a `retrieveRecords()` function to fetch the items in the `sessionStorage`. This function is outlined below. As noted, the `getItem()` requires a key to retrieve a specific object or value.
+
+```javascript
+function retrieveRecords(){ //retrieves items in the sessionStorage
+        console.log("retrive records");
+        var records = window.sessionStorage.getItem('car');
+        var paragraph = document.createElement("p");
+        var infor = document.createTextNode(records);
+        paragraph.appendChild(infor);
+        var element = document.getElementById("retrieve");
+        element.appendChild(paragraph);
+}
+```
+
+Just like the `getItem()`, the `removeItem()` method requires a key. In this example, the key is the `car`.
+
+```javascript
+function removeItem(){//deletes item from sessionStorage
+        sessionStorage.removeItem('car')
+        console.log("remove items");
+}
+```
+
+The clearStorage function will delete all items in the sessionStorage when called. Here is the required code snippet.
+
+```javascript
+function clearStorage(){ //clears the entire sessionStorage
+        sessionStorage.clear()
+        console.log("clear records");
+}
+```
+
+Here is the entire code in the main.js file.
+
+```javascript
+
+function store(){ //stores items in the local storage
+    var brand = document.getElementById('carBrand').value;
+    var price = document.getElementById('carPrice').value;
+
+    const car = {
+        brand: brand,
+        price: price,
+    }
+    
+    window.sessionStorage.setItem('car',JSON.stringify(car));  
+    //converting object to string
+}
+function retrieveRecords(){ //retrieves items in the sessionStorage
+        console.log("retrive records");
+        var records = window.sessionStorage.getItem('car');
+        var paragraph = document.createElement("p");
+        var infor = document.createTextNode(records);
+        paragraph.appendChild(infor);
+        var element = document.getElementById("retrieve");
+        element.appendChild(paragraph);
+}
+function removeItem(){//deletes item from sessionStorage
+        sessionStorage.removeItem('car')
+        console.log("remove items");
+}
+
+function clearStorage(){ //clears the entire sessionStorage
+        sessionStorage.clear()
+        console.log("clear records");
+}
+
+window.onload =function(){ //ensures the page is loaded before functions are executed.
+        document.getElementById("carForm").onsubmit = store
+        document.getElementById("clearButton").onclick = clearStorage
+        document.getElementById("removeButton").onclick = removeItem
+        document.getElementById("retrieveButton").onclick = retrieveRecords
+}
+```
+### Results
+You can follow the video below to test the web page we have developed in this tutorial.
```

Comment:
You should mention that the object stored is related to the most recent car's details submitted. You may also guide the user to modify the code to store a list in the sessionStorage.